src/js/angular/resources/chart-models/heap-memory-chart.js

```diff
@@ -0,0 +1,50 @@
+import {ChartData} from "./chart-data";
+
+export class HeapMemoryChart extends ChartData {
+    chartSetup() {
+        this.chartOptions.chart.color = d3.scale.category10().range();
+        this.chartOptions.title = {
+            className: 'chart-additional-info'
+        };
+    }
+    createDataHolder() {
+        return [{
+            key: this.translateService.instant('resource.memory.committed'),
+            values: []
+        }, {
+            key: this.translateService.instant('resource.memory.used'),
+            area: 'true',
+            values: []
+        }];
+    }
+    addNewData(timestamp, data) {
+        const [committed, used] = this.data;
```

Comment:
Why `this.data` and not just `data`?

Reply:
I should probably change the name of `this.data` to `dataHolder` or something so it is not confused with the data from the response.

`this.data` is where I store the properties. `data` is the data from the response.

Reply:
I changed `this.data` to `this.dataHolder`